protostar/compiler/project_compiler.py

```diff
@@ -0,0 +1,136 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Union
+
+from starkware.cairo.lang.compiler.preprocessor.preprocessor_error import (
+    PreprocessorError,
+)
+from starkware.cairo.lang.vm.vm_exceptions import VmException
+from starkware.starknet.services.api.contract_class import ContractClass
+from starkware.starkware_utils.error_handling import StarkException
+
+from protostar.compiler.compiled_contract_writer import CompiledContractWriter
+from protostar.protostar_exception import ProtostarException
+from protostar.protostar_toml.protostar_contracts_section import (
+    ProtostarContractsSection,
+)
+from protostar.utils.compiler.pass_managers import StarknetPassManagerFactory
+from protostar.utils.starknet_compilation import CompilerConfig, StarknetCompiler
+
+from .project_cairo_path_builder import ProjectCairoPathBuilder
+
+ContractName = str
+ContractSourcePath = Path
+ContractIdentifier = Union[ContractName, ContractSourcePath]
+
+
+class ProjectCompiler:
+    @dataclass
+    class Config:
+        relative_cairo_path: List[Path]
+        debugging_info_attached: bool = False
+        hint_validation_disabled: bool = False
```

Comment:
Could this be renamed to `ProjectCompilerConfig` and put outside the `ProjectCompiler` class? I find it unergonomic to use nested classes because they are hard to find when grepping symbols in the source code/autocompletion.

Reply:
I tend to use nested classes because they don't have to be imported. I can put it outside.

Reply:
do you manage imports manually when writing code? because I'm used to have good auto importer running in my code editor 😅